application.go

```diff
@@ -25,18 +26,23 @@ type State struct {
 }
 
 type Application interface {
-	Setup(cfgs ...any) func(cmd *cobra.Command, args []string) error
+	AddConfigs(cfgs ...any)
+	Config() *Config
```

Comment:
is a reference really required? the original goal is that what was passed in should remain immutable.